docs/standards/javascript.md

```diff
@@ -0,0 +1,203 @@
+# Javascript Reference
+
+## Policy
+
+### Coding standards
+
+Javascript code should follow the same standards as CiviCRM's [PHP coding standards](https://docs.civicrm.org/dev/en/master/standards/php/). You can use inline tools like [JSHint](http://jshint.com/) to help with the linting of javascript files. If you have buildkit installed JSHint is included as part of [CiviLint](https://docs.civicrm.org/dev/en/master/tools/civilint/). Adding hints to your code will enable jshint to check the standards: e.g.
+
+Example to tell JSHint Tell jsHint about any globals in use:
+```javascript
+/*global CRM, ts */
+```
+### globals
+
+Declaring a global variable/function is a bad practice in Javascript unless absolutely necessary. Your code should never create globals. In the rare cases that you need to declare variables or functions outside the local scope of your closure, create a namespace within the CRM object
+
+```javascript
+CRM.myStuff = {foo: 'bar'}.
+```
+
+Note: due to legacy code written before these standards were adopted, CiviCRM still has quite a few other global variables and functions. Aside from the 2 listed below, they all need to be removed from the global scope. You can help!
+
+### Location
+
+Javascript code should only really be found in three main locations
+
+1 Inline scripts should be included in smarty .tpl template files. This should only be done where its limited to a specific page or form. Inline js must be enclosed in smarty {literal} tags
```

Comment:
Not certain if `mkdocs` is smarter, but Github markdown doesn't quite recognize this list. Maybe needs to be "` 1. `" (`<space><one><dot><space>` )